### ISSUE TITLE
New formula for bib-tool doesn't work on linux

### ISSUE THREAD

**Issue by adamliter** [Homebrew/linuxbrew#456]
In the newest version of BibTool (2.60), there is a problem with the tarball. It has duplicate files. It works perfectly fine to unarchive the tarball with gnutar but not with bsdtar. The latter doesn't like duplicate files. In order to work around this, the new bib-tool formula does:

```ruby
class BibToolDownloadStrategy < CurlDownloadStrategy
  def stage
     with_system_path { safe_system 'tar', 'xqf', cached_location, 'BibTool/doc/bibtool.tex' }
     with_system_path { safe_system 'tar', 'xf', cached_location, '--exclude', 'BibTool/doc/bibtool.tex' }
     chdir
  end
end
```

That is, it ends up running:

```sh
tar xqf bib-tool-2.60.tar.gz BibTool/doc/bibtool.tex
tar xf --exclude bib-tool-2.60.tar.gz BibTool/doc/bibtool.tex
```
(See https://github.com/Homebrew/homebrew/issues/40559)

However, this is a problem with gnutar (which is what is installed on most (all?) linux machines) since `-q` (AKA `--fast-read`) is only implemented in bsdtar.

I don't know how the formula maintenance works between Homebrew and Linuxbrew, but I assume you like to keep the formulas the same, correct?

If so, something like the following will need to be done.

```ruby
class BibToolDownloadStrategy < CurlDownloadStrategy
  if `tar --version`.match 'bsdtar'
    def stage
      with_system_path { safe_system 'tar', 'xqf', cached_location, 'BibTool/doc/bibtool.tex' }
      with_system_path { safe_system 'tar', 'xf', cached_location, '--exclude', 'BibTool/doc/bibtool.tex' }
      chdir
    end
  else
    def stage
      with_system_path { safe_system 'tar', 'xf', cached_location }
      chdir
    end
  end
end
```

**Comment by adamliter** [comment]
I think https://github.com/adamliter/homebrew/commit/6dd00da44e4532a69b6b8e722638858fe8eb50fc should fix the problem. I can test it on a virtual Linux machine when I get home later today to make sure. But the code should work.

Should this be submitted as a pull request to Hombrew/homebrew, though, or to Homebrew/linuxbrew? Like I said, I'm not sure what the policies are with respect to the maintenance of formulas across Homebrew and Linuxbrew.

**Comment by adamliter** [comment]
I just tested this on a virtual linux machine, and it seems to work.

**Comment by sjackman** [comment]
Since this is a wacky upstream bug, I'd prefer to just wait for the upstream fix if they're responsive. Is there an upstream bug filed?

**Comment by adamliter** [comment]
@sjackman Yes, the tarball has been fixed, but [the maintainer doesn't want to do a new release just for this fix](https://github.com/Homebrew/homebrew/issues/40559#issuecomment-110873643), and it's unclear when the next version will actually be released. Probably not anytime all that soon.

As it stands, if somebody runs `brew install bib-tool` with Linuxbrew, they will get:

```
==> Downloading http://www.gerd-neugebauer.de/software/TeX/BibTool/BibTool-2.60.
################################################################## 100.0%

tar: invalid option -- 'q'
Try 'tar --help' or 'tar --usage' for more information.
Error: Failure while executing: tar xqf /home/USERNAME/.cache/Homebrew/bib-tool-2.60.tar.gz BibTool/doc/bibtool.tex
```
This is my fault for not thinking about this when I fixed the problem with the workaround in https://github.com/Homebrew/homebrew/commit/bab2916f7afe8aede0ae2e188018fcb529f760f2. The workaround there only works with bsdtar, which is why that error is thrown when you try to install with Linuxbrew.

I agree that this should be fixed by the maintainer, and it's unfortunate that he won't do a new release to fix the tarball, but it might be better to fix the issue now in case anyone does try to install BibTool with Linuxbrew in the meantime while we wait for the next release of BibTool.

But, if you still want to wait, I understand.

**Comment by sjackman** [comment]
That's quite annoying that upstream won't release a not-broken tarball. I'll merge a PR to fix this. Do move the `if` statement inside the `def stage` statement.

**Comment by adamliter** [comment]
Okay, thanks @sjackman ! So I should move the `if` statement and then should I submit the pull request to Homebrew/hombrew or Homebrew/linuxbrew? Or were you going to do both of those things?

**Comment by adamliter** [comment]
I did what you suggested in https://github.com/adamliter/homebrew/commit/7d298c83ff29ccf5d3c9954a536a7564150a8a7d. Should I open a pull request to Homebrew/homebrew?

**Comment by sjackman** [comment]
I doubt that Homebrew/homebrew would merge this patch. Open the PR at Homebrew/linuxbrew.

**Comment by adamliter** [comment]
@sjackman I just opened the pull request. Thanks for the comments and suggestions on the commit.

**Comment by sjackman** [comment]
Fixed by #457.
